### PR TITLE
Handle new model cache states

### PR DESCRIPTION
**Pull request by kulyk**
Part of #22538

Adding FE support for new model cache states. People can cache their model query results and the process of caching can end up in a few states.

## Background
We used to have:

* `refreshing` — refreshing cache right now
* `persisted` — refreshed cache, not doing anything with it now
* `error` — something went wrong while refreshing cache

Now we also have:

* `creating` — creating the first cache, there's nothing to refresh yet
* `deletable` — happens when you e.g. turn off model persistence for the whole database. All cache records get marked as `deletable` and will be eventually cleaned up
* `off` — happens when a user manually disables caching for a specific model

Overall, the state sequence is: `creating` → `persisted` → `refreshing` → `persisted` / `error` → ∞ → `off` / `deletable`

## Changes

We display cache states in two places: the model details sidebar on the model page + Tools page listing all the cache states.

**Model details sidebar**

1. Not going to show the cache section at all if it's `off` or `deletable`
2. Will show the "Caching in progress" message in the `creating` state
3. Will only show the "refresh" button in `persisted` or `error` states

**Tools cache info listing**

1. Added an explicit empty state
2. Not going to show info records in `deletable` state
3. Will show "Off" status message for `off` state
4. Will show "Caching in progress" status message for `creating` state
5. Will only show the "refresh" button in `persisted` or `error` states

### CODE REVIEW

**Review by luizarakaki — APPROVED** [review]
Overall looks good to me, but there is an issue:

There is an issue with the `creating` state. When users enable cache, it waits until the next schedule to create the first persistence. So the `caching in progress` message doesn't make sense all the time. Maybe we need a `scheduled` status for the first creation.

**Comment by kulyk** [comment]
@luizarakaki yeah, seems like `creating` is exactly that, my bad! What would be the better copy here? Something like "Queued"? We can also use the refresh schedule settings to show when the next refresh is going to happen.

**Comment by luizarakaki** [comment]
Yeah `queued` makes sense! I also like the idea of showing when the next refresh is going to happen.

File: enterprise/frontend/src/metabase-enterprise/model_persistence/components/ModelCacheManagementSection/ModelCacheManagementSection.tsx [hunk 1]
```diff
@@ -28,9 +29,15 @@ type LoaderRenderProps = {
 };
 
 function getStatusMessage(job: ModelCacheRefreshStatus) {
+  if (job.state === "off") {
+    return `Caching is turned off`;
```
Comment (daltojohnso) [inline enterprise/frontend/src/metabase-enterprise/model_persistence/components/ModelCacheManagementSection/ModelCacheManagementSection.tsx:33]: missing a `t` here

**Review by daltojohnso — APPROVED** [review]

**Pull request merged**